bson/value_writer.go

```diff
@@ -33,6 +33,29 @@ func putValueWriter(vw *valueWriter) {
 	}
 }
 
+var documentWriterPool = sync.Pool{
+	New: func() interface{} {
+		return NewDocumentWriter(nil)
```

Comment:
Can we call `newDocumentWriter()` because it returns a `*valueWriter` for syntactic clarity?